make_queue/models/models.py

```diff
@@ -99,6 +100,18 @@ class Machine(models.Model):
 
     status = models.CharField(max_length=2, choices=STATUS_CHOICES, verbose_name=_("Status"), default=AVAILABLE)
     name = models.CharField(max_length=30, unique=True, verbose_name=_("Name"))
+    stream_name = models.CharField(
+        max_length=30,
+        verbose_name=_("Stream Name"),
+        default=None, blank=True, null=True,
+        validators=[
+            RegexValidator(
+                regex="^[a-zA-Z0-9_-]+$",
+                message=_("Stream name can only consist of english letters, numbers, hyphens or underscores."),
+                code="NOT_URL_SAFE"
+            )
+        ]
+    )
```

Comment:
I would like a custom text, which is why I created my own.

Reply:
I would encourage to use `slug_re` (which `validate_slug` uses) as the `regex` kwarg, so that we don't have to redefine a functionally equal regex string 🙂

Reply:
updated!